app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -204,12 +224,47 @@ class LocationSensorManager : BroadcastReceiver(), SensorManager {
 
     private fun handleGeoUpdate(intent: Intent) {
         Log.d(TAG, "Received geofence update.")
+        if (!isEnabled(latestContext, zoneLocation.id)) {
+            isZoneLocationSetup = false
+            Log.w(TAG, "Unregistering geofences as zone tracking is disabled and intent was received")
+            removeGeofenceUpdateRequests()
+            return
+        }
         val geofencingEvent = GeofencingEvent.fromIntent(intent)
         if (geofencingEvent.hasError()) {
             Log.e(TAG, "Error getting geofence broadcast status code: ${geofencingEvent.errorCode}")
             return
         }
 
+        val validGeofencingEvents = listOf(Geofence.GEOFENCE_TRANSITION_ENTER, Geofence.GEOFENCE_TRANSITION_EXIT)
+        if (geofencingEvent.geofenceTransition in validGeofencingEvents) {
+            when (geofencingEvent.geofenceTransition) {
+                Geofence.GEOFENCE_TRANSITION_ENTER -> zoneStatusEvent = "android.zone_entered"
+                Geofence.GEOFENCE_TRANSITION_EXIT -> zoneStatusEvent = "android.zone_exited"
+            }
+            val zoneAttr = geofencingEvent.triggeringLocation.extras.keySet()
+                .map { it to geofencingEvent.triggeringLocation.extras.get(it) }
+                .toMap()
+                .plus("accuracy" to geofencingEvent.triggeringLocation.accuracy)
+                .plus("altitude" to geofencingEvent.triggeringLocation.altitude)
+                .plus("bearing" to geofencingEvent.triggeringLocation.bearing)
+                .plus("latitude" to geofencingEvent.triggeringLocation.latitude)
+                .plus("longitude" to geofencingEvent.triggeringLocation.longitude)
+                .plus("provider" to geofencingEvent.triggeringLocation.provider)
+                .plus("time" to geofencingEvent.triggeringLocation.time)
+                .plus("zone" to geofencingEvent.triggeringGeofences[0].requestId)
```

Comment:
The extras is only adding `verticalAccuracy` actually. I can switch it to `mapOf` if next update :)

Reply:
I just noticed vertical accuracy is only for certain SDKs maybe its best to not include it then I dont know how important it would be to the zone data.